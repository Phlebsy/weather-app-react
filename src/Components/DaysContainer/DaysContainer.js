import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DaysCard from '../DaysCard/DaysCard';
import { changeDay } from '../../context/actions';
import OwlCarousel from 'react-owl-carousel';

// DaysContainer builds the five-six DayCard components and places them into the carousel
// Logic is from actions and passed into children components to change active forecasts
export default class DaysContainer extends Component {
  render() {
    const { days, dispatch } = this.props;

    let dayCards = days.map((day, index) => {
      if (day['fullDay']) {
        return (
          <DaysCard
            onClick={changeDay.bind(this, dispatch, index)}
            day={day}
            key={`day-${index}`}
          />
        );
      }
      return undefined;
    });

    dayCards = dayCards.filter(day => day !== undefined); // Fix for days with no forecasts left

    return (
      <OwlCarousel
        dots={false}
        slideBy={3}
        nav={true}
        className="owl-theme"
        margin={10}
        items={3}
      >
        {dayCards}
      </OwlCarousel>
    );
  }
}

DaysContainer.propTypes = {
  days: PropTypes.array.isRequired,
  dispatch: PropTypes.func,
  dayIndex: PropTypes.number
};
