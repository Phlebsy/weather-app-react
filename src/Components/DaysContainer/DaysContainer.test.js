import React from 'react';
import ReactDOM from 'react-dom';
import DaysContainer from './DaysContainer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const days = [
  {
    avg: 276.25,
    fullDay: 'Tue - Dec 4',
    weather: 'clear sky'
  },
  {
    avg: 272.754625,
    fullDay: 'Wed - Dec 5',
    weather: 'clear sky'
  }
];

describe('<DaysContainer />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DaysContainer days={days} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<DaysContainer days={days} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
