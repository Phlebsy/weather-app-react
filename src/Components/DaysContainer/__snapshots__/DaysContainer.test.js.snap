// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaysContainer /> matches the snapshot 1`] = `
<ReactOwlCarousel
  className="owl-theme"
  dots={false}
  items={3}
  margin={10}
  nav={true}
  slideBy={3}
>
  <DaysCard
    day={
      Object {
        "avg": 276.25,
        "fullDay": "Tue - Dec 4",
        "weather": "clear sky",
      }
    }
    key="day-0"
    onClick={[Function]}
  />
  <DaysCard
    day={
      Object {
        "avg": 272.754625,
        "fullDay": "Wed - Dec 5",
        "weather": "clear sky",
      }
    }
    key="day-1"
    onClick={[Function]}
  />
</ReactOwlCarousel>
`;
